uk.co.compucorp.civicrm.hrcore/gulpfile.js

```diff
@@ -46,6 +47,31 @@ var Promise = require('es6-promise').Promise;
     });
   });
 
+  /**
+   * Gets a map of contact roles and IDs
+   *
+   * @return {Promise} resolved with {Object}
+   */
+  function getRolesAndIDs () {
+    return new Promise(function (resolve) {
+      var rolesAndIDs = {};
+
+      exec('cv api contact.get', {maxBuffer: 1024 * 1000}, function (err, result) {
```

Comment:
use a more specific query
```
cv api Contact.get sequential=1 return="display_name" display_name="civihr_" contact_type="Individual"
```

Reply:
Here is an example of the contact returned by the command that is currently there:

```js
{ contact_id: '204',
  contact_type: 'Individual',
  contact_sub_type: '',
  sort_name: 'Pavlov, Igor', // after wizard
  display_name: 'Igor Pavlov', // after wizard
  do_not_email: '0',
  do_not_phone: '0',
  do_not_mail: '0',
  do_not_sms: '0',
  do_not_trade: '0',
  is_opt_out: '0',
  legal_identifier: '',
  external_identifier: '',
  nick_name: 'Igor', // after wizard
  legal_name: '',
  image_URL: '',
  preferred_communication_method: '',
  languages: 'English (United Kingdom)',
  preferred_language: 'en_GB',
  preferred_mail_format: 'Both',
  first_name: 'Igor', // after wizard
  middle_name: '',
  last_name: 'Pavlov', // after wizard
  individual_prefix: '',
  prefix_id: '',
  individual_suffix: '',
  suffix_id: '',
  formal_title: '',
  communication_style: '',
  communication_style_id: '',
  job_title: '',
  gender: '',
  gender_id: '',
  birth_date: '2018-01-10', // after wizard
  is_deceased: '0',
  deceased_date: '',
  household_name: '',
  organization_name: '',
  sic_code: '',
  contact_is_deleted: '0',
  current_employer: '',
  address_id: '184', // after wizard?
  street_address: 'Flat', // after wizard
  supplemental_address_1: 'Street', // after wizard
  supplemental_address_2: '',
  supplemental_address_3: '',
  city: 'London',
  postal_code_suffix: '',
  postal_code: '55555', // after wizard
  geo_code_1: '',
  geo_code_2: '',
  state_province_id: '',
  country_id: '1226', // after wizard
  phone_id: '',
  phone_type_id: '',
  phone: '',
  email_id: '198',
  email: 'civihr_staff@compucorp.co.uk',
  on_hold: '0',
  im_id: '',
  provider_id: '',
  im: '',
  worldregion_id: '1',
  world_region: 'Europe and Central Asia',
  state_province_name: '',
  state_province: '',
  country: 'United Kingdom',
  id: '204' }
```
As you can see, we cannot rely on the display_name. However, the `sequential=1 contact_type="Individual"` and additionally `return="email,contact_id"` will help. Thanks.